lib/components/src/ScrollBar/ScrollArea.tsx

```diff
@@ -0,0 +1,344 @@
+import { useDOMRect } from '@storybook/addons';
+import React, { FC, useCallback, useEffect, useRef, useState } from 'react';
+import { TrackHorizontal } from './components/TrackHorizontal';
+import { TrackVertical } from './components/TrackVertical';
+import {
+  SLIDER_DEFAULT_FADEOUT,
+  SLIDER_DEFAULT_OPACITY,
+  SLIDER_DEFAULT_PADDING,
+  SLIDER_DEFAULT_SIZE,
+} from './const';
+import * as Styled from './styled';
+import { ScrollAreaProps, ScrollAreaState, ScrollAreaStateItem } from './types';
+import { getHorizontalState } from './utils/get-horiztontal-state';
+import { getVerticalState } from './utils/get-vertical-state';
+
+export const ScrollArea: FC<ScrollAreaProps> = ({
+  absolute,
+  children,
+  horizontal: enableHorizontal = true,
+  horizontalPosition = 'bottom',
+  showOn = 'hover',
+  sliderColor,
+  sliderFadeout = SLIDER_DEFAULT_FADEOUT,
+  sliderOpacity = SLIDER_DEFAULT_OPACITY,
+  sliderPadding = SLIDER_DEFAULT_PADDING,
+  sliderSize = SLIDER_DEFAULT_SIZE,
+  sliderType,
+  vertical: enableVertical = true,
+  verticalPosition = 'right',
+  onScroll,
+  ContentProps = {},
```

Comment:
Why pascal case?

Reply:
I like to differentiate betweeen props that is for the wrapper and props that are for support components

Reply:
If this is an approach not to the teams liking I will change off course

Reply:
definitely prefer camelCase!

Reply:
fixed